Kubernetes/linux/manifest/generate.py

```diff
@@ -0,0 +1,260 @@
+spec:
+  hostNetwork: true
+  containers:
+    - name: "kube-apiserver"
+      image: "gcr.io/google_containers/hyperkube-amd64:v1.7.3"
+      command:
+        - "/hyperkube"
+        - "apiserver"
+        - "--admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,ResourceQuota"
+        - "--address=0.0.0.0"
+        - "--allow-privileged"
+        - "--insecure-port=8080"
+        - "--secure-port=443"
+        - "--service-cluster-ip-range=11.0.0.0/16"
+        - "--etcd-servers=http://127.0.0.1:2379"
+        - "--etcd-quorum-read=true"
+        - "--advertise-address=$MASTERIP"
+        - "--tls-cert-file=/etc/kubernetes/certs/apiserver.pem"
+        - "--tls-private-key-file=/etc/kubernetes/certs/apiserver-key.pem"
+        - "--client-ca-file=/etc/kubernetes/certs/ca.pem"
+        - "--service-account-key-file=/etc/kubernetes/certs/apiserver-key.pem"
+        - "--storage-backend=etcd2"
+        - "--v=4"
+      ports:
+        - name: "https"
+          containerPort: 443
+          hostPort: 443
+        - name: "local"
+          containerPort: 8080
+          hostPort: 8080
+      volumeMounts:
+        - name: "etc-kubernetes"
+          mountPath: "/etc/kubernetes"
+        - name: "var-lib-kubelet"
+          mountPath: "/var/lib/kubelet"
+  volumes:
+    - name: "etc-kubernetes"
+      hostPath:
+        path: "$HOME/kube"
+    - name: "var-lib-kubelet"
+      hostPath:
+        path: "$HOME/kube/kubelet"
+""",
+    "kube-control-manager.yaml": """apiVersion: "v1"
+kind: "Pod"
+metadata:
+  name: "kube-controller-manager"
+  namespace: "kube-system"
+  labels:
+    tier: control-plane
+    component: kube-controller-manager
+spec:
+  hostNetwork: true
+  containers:
+    - name: "kube-controller-manager"
+      image: "gcr.io/google_containers/hyperkube-amd64:v1.7.3"
+      command:
+        - "/hyperkube"
+        - "controller-manager"
+        - "--kubeconfig=/var/lib/kubelet/config"
+        - "--allocate-node-cidrs=True"
+        - "--cluster-cidr=$CLUSTER"
+        - "--cluster-name=kubernetes"
+        - "--root-ca-file=/etc/kubernetes/certs/ca.pem"
+        - "--cluster-signing-cert-file=/etc/kubernetes/certs/ca.pem"
+        - "--cluster-signing-key-file=/etc/kubernetes/certs/ca-key.pem"
+        - "--service-account-private-key-file=/etc/kubernetes/certs/apiserver-key.pem"
+        - "--leader-elect=true"
+        - "--v=2"
+      volumeMounts:
+        - name: "etc-kubernetes"
+          mountPath: "/etc/kubernetes"
+        - name: "var-lib-kubelet"
+          mountPath: "/var/lib/kubelet"
+        - name: "varlog"
+          mountPath: "/var/log"
+  volumes:
+    - name: "etc-kubernetes"
+      hostPath:
+        path: "$HOME/kube"
+    - name: "var-lib-kubelet"
+      hostPath:
+        path: "$HOME/kube/kubelet"
+    - name: "varlog"
+      hostPath:
+        path: "$HOME/kube/log/kube-controller-manager"
+""",
+    "kube-scheduler.yaml": """
+    apiVersion: "v1"

```

Comment:
Same problem here, the whitespace before apiVersion for kube-scheduler.yaml is causing problems

Reply:
Dang, I thought I fixed that already before committing. Thanks!